packages/babel-parser/src/plugin-utils.js

```diff
@@ -74,16 +75,31 @@ export function validatePlugins(plugins: PluginList) {
     throw new Error("Cannot combine placeholders and v8intrinsic plugins.");
   }
 
-  if (
-    hasPlugin(plugins, "pipelineOperator") &&
-    !PIPELINE_PROPOSALS.includes(
-      getPluginOption(plugins, "pipelineOperator", "proposal"),
-    )
-  ) {
-    throw new Error(
-      "'pipelineOperator' requires 'proposal' option whose value should be one of: " +
-        PIPELINE_PROPOSALS.map(p => `'${p}'`).join(", "),
-    );
+  if (hasPlugin(plugins, "pipelineOperator")) {
+    const proposal = getPluginOption(plugins, "pipelineOperator", "proposal");
+
+    if (!PIPELINE_PROPOSALS.includes(proposal)) {
+      const proposalList = PIPELINE_PROPOSALS.map(p => `"${p}"`).join(", ");
+      throw new Error(
+        `"pipelineOperator" requires "proposal" option whose value must be one of: ${proposalList}.`,
+      );
+    }
+
+    if (proposal === "hack") {
+      const topicToken = getPluginOption(
+        plugins,
+        "pipelineOperator",
+        "topicToken",
+      );
+
+      if (!TOPIC_TOKENS.includes(topicToken)) {
+        const tokenList = TOPIC_TOKENS.map(t => `"${t}"`).join(", ");
+
+        throw new Error(
+          `"pipelineOperator" in "proposal": "hack" mode also requires a "topicToken" option whose value must be one of: ${tokenList}.`,
+        );
+      }
```

Comment:
Per https://github.com/js-choi/proposal-hack-pipes/issues/3 we should also throw if `topicToken` is `#` and we have `"paritialApplication"` plugin.